src/indexed_crate.rs

```diff
@@ -1260,6 +1260,31 @@ mod tests {
             }
         }
 
+        #[test]
+        fn trait_types() {
+            let test_crate = "traits_with_types";
+
+            let rustdoc = load_pregenerated_rustdoc(test_crate);
+            let indexed_crate = IndexedCrate::new(&rustdoc);
+
+            let associated_types = indexed_crate
+                .inner
+                .index
+                .values()
+                .filter_map(|x| {
+                    let ItemEnum::AssocType{default, ..} = &x.inner else {return None};
+                    Some((&x.name, default))
+                })
+                .fold(std::collections::HashMap::new(), |mut acc, el| {
+                    let Some(v) = el.0 else { return acc };
+                    acc.insert(v.clone(), el.1.clone());
+                    acc
+                });
+
+            assert!(&associated_types["DeserializedType"].is_none());
+            assert!(&associated_types["SerializedType"].is_some());
+        }
```

Comment:
This test is actually in the wrong submodule — it's in the `reexports` submodule even though it doesn't test re-exports at all.

We actually probably don't need this test at all, because it just checks something that's you already added a test for in `src/adapter/tests.rs`.

I'd just delete it.